### ISSUE TITLE
Review and update recommendations regarding secure deletion

### ISSUE THREAD

**Issue by eloquence** [freedomofpress/securedrop-docs#563]
## Describe the change
We should ensure our recommendations regarding secure deletion make sense for USB/SSD deletion, and that we help journalists mitigate risks with files that could be recovered by an adversary who gains access to a drive and passphrase.

This is especially true for the export device, which holds files in decrypted form (the drive itself is meant to be encrypted; our docs recommend VeraCrypt for this device, as it may be opened on non-Linux machines).

For example, we could recommend a regular drive-level overwrite operation for folks who routinely re-use a single export device -- but we'd need to ensure that such a recommendation actually meaningfully mitigates risk.

## How will this impact [SecureDrop users](https://github.com/freedomofpress/securedrop-ux/wiki/Users)?

- mitigate risk from search & seizures

## Additional context

- Tails 6 removed support for ["secure wipe"](https://tails.net/doc/encryption_and_privacy/secure_deletion/) functionality, since the reliability of such wipe operations on a per-file basis cannot be guaranteed on USB/SSD devices.
- #558 updated the docs accordingly and no longer recommends "securely" wiping individual files

**Comment by ChumOfChance** [comment]
The instructions for decommissioning SecureDrop ([step 6](https://docs.securedrop.org/en/stable/admin/maintenance/decommission.html)) are good for LUKS-encrypted USB drives and based on my understanding of the best practice: destroy the LUKS header, which can be most easily done by simply reformatting the disk as a new LUKS-encrypted disk. See https://wiki.gentoo.org/wiki/Secure_wipe

However, these instructions incorrectly state that all USB drives involved in SecureDrop use LUKS encryption. Some users will be using VeraCrypt based drives as their Export Device. Regardless, reformatting all drives with LUKS should make the contents irrecoverable. 

In `admin/installation/set_up_transfer_and_export_device.html`, we do state "it is important that the contents of the Transfer Device and the Export Device are always wiped after a copy operation is completed." This should definitely be changed. 

I think the best recommendation to mitigate the risk from search & seizures with regards to files on USB drives that could be recovered is to regularly reformat the Transfer and Export devices, using the same passphrase. When reformatting a LUKS drive, even with the same passphrase, the master key is changed. Doing this on every use is not a realistic recommendation, but this could be a recommended on a regular cycle to minimize the fallout of a potential seizure, or whenever it is desirable to "wipe" the contents of a drive. The passphrase doesn't change, so there is minimal impact for journalists using the drives.

From https://kernel.googlesource.com/pub/scm/utils/cryptsetup/cryptsetup/+/dae91fd9ec2f7d89e8d4c6a39a0ef74040f3bf32/FAQ.md :

> LUKS PASSPHRASE IS NOT THE MASTER KEY: The LUKS passphrase is not used in deriving the master key. It is used in decrypting a master key that is randomly selected on header creation. This means that if you create a new LUKS header on top of an old one with exactly the same parameters and exactly the same passphrase as the old one, it will still have a different master key and your data will be permanently lost.

**Comment by ChumOfChance** [comment]
VeraCrypt works similarly, in that creating a new volume from scratch (even with the same passphrase) will create a new master key in the volume header. With the old master key destroyed, the data on the drive won't be recoverable anymore. 

https://veracrypt.io/en/Changing%20Passwords%20and%20Keyfiles.html

**Comment by ChumOfChance** [comment]
A good reference is the Tails documentation: https://tails.net/doc/encryption_and_privacy/secure_deletion/index.en.html#spare

**Comment by ChumOfChance** [comment]
Additionally, we state "If you follow this recommendation [to use encrypted Export and Transfer USB flash drives], it is important that the contents of the Transfer Device and the Export Device are always wiped after a copy operation is completed." (https://docs.securedrop.org/en/stable/admin/installation/set_up_transfer_and_export_device.html#choose-media-types-and-encryption) 

But, we do not provide instructions on how to do so, besides in the decommissioning section. For journalists, we say "As a security precaution, we recommend deleting the files on the Export Device after each copy operation." (https://docs.securedrop.org/en/stable/journalist/documents.html#decrypting-and-preparing-to-publish) but that does not provide a "secure deletion", and sometimes a journalist may want to explicitly want to take extra measures to securely remove traces of a file on a USB flash drive.

**Comment by legoktm** [comment]
> Doing this on every use is not a realistic recommendation, but this could be a recommended on a regular cycle to minimize the fallout of a potential seizure, or whenever it is desirable to "wipe" the contents of a drive.

Unless/until we have an automated way to provision a USB drive (there's probably an issue for this somewhere), I agree that it's unrealistic. 

AIUI there's no actual way to "securely" delete these because of physical hardware limitations short of overwriting the entire drive, which will take a long time and also kill them faster. Is that still correct?

My current thinking is to split the threat model into two:
1) General deletion by reformatting the drive (destroying luks/veracrypt keys), makes it unrecoverable by most(??) attackers and generally good practice. I don't know on what cycle to recommend this or where in the workflow we'd automate this.
2) If you really really want no one to ever recover a document, physically destroy the drive.

**Comment by ChumOfChance** [comment]
Even overwriting a USB flash drive isn't guaranteed to work for various technical hardware reasons. So, the best recommendation is to start with an encrypted drive, and reformat the drive. Reformatting the drive destroys/overwrites the encryption headers, which (as I understand it) is more reliable than generally overwriting the entire drive to make any content unrecoverable. 

So to update our recommendations, I would suggest re-formatting/re-encrypting the Export Devices as a best practice, to be done with some regularity (to be determined by the user), providing the steps to do so (by linking to where we already have created Export Devices documented). With the caveat that in the case of needing to completely assure a file is unrecoverable the physical device must be destroyed.

**Comment by redshiftzero** [comment]
Agreed - re-encrypting the drive to destroy the previous encryption headers is preferred, from NIST document [here](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-88r2.pdf) which may be worth citing (I looked at this to see what the latest recommendations were here):

<img width="869" height="194" alt="Image" src="https://github.com/user-attachments/assets/0d806023-ceaa-45e0-bed7-47decb741f88" />

For the highest assurance (e.g. when taking a drive out of active use), destruction is preferred.

**Comment by ChumOfChance** [comment]
Indeed. The CCCS guidance and explanations are equivalent (and a bit more readable imo): https://www.cyber.gc.ca/#a22. They refer to this process as "Crypto Erase".

I feel equipped to convert this guidance into the relevant PR!
